src/hal/src/adapter.rs

```diff
@@ -60,7 +60,7 @@ pub trait PhysicalDevice<B: Backend>: Sized {
     fn open(self, Vec<(B::QueueFamily, Vec<QueuePriority>)>) -> Gpu<B>;
 
     ///
-    fn format_properties(&self, format::Format) -> format::Properties;
+    fn format_properties(&self, Option<format::Format>) -> format::Properties;
```

Comment:
does this function even make sense to call with `None`? I'd like to have the format passed here straight

Reply:
Yes, uniform/storage texel buffers with `Format::Undefined` are allowed. (also called untyped buffers).
It's also handled inside the open AMD drivers.

Reply:
I guess that it's required due to `create_buffer_view`. We need check the supported buffer features for `VK_FORMAT_UNDEFINED`.